server/datastore/mysql/hosts.go

```diff
@@ -62,6 +62,26 @@ func (d *Datastore) NewHost(ctx context.Context, host *fleet.Host) (*fleet.Host,
 	return host, nil
 }
 
+func (d *Datastore) SerialSaveHost(ctx context.Context, host *fleet.Host) error {
+	errCh := make(chan error)
+	defer close(errCh)
+	select {
+	case <-ctx.Done():
+		return nil
+	case d.writeCh <- itemToWrite{
+		ctx:   ctx,
+		errCh: errCh,
+		item:  host,
+	}:
+		select {
+		case <-ctx.Done():
+			return nil
```

Comment:
Should probably return `ctx.Err()` here, it's not a successful call and it will let the caller know it failed due to deadline exceeded.

Reply:
Well, in this case, the caller is osquery, and it'll retry later anyway.

Reply:
Ok but that's still a failed call, I don't think the fact that it will be retried should change that.

Reply:
Fair.